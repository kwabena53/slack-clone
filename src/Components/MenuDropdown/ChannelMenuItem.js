
import '../../App.css'
import './MenuDropdown.css'


const ChannelMenuItem = ({name, ...rest})=>{
    return(
            <button {...rest} className="menuItem">
                <div className="subMenu">
                  <span className="menuIcon"> #</span>
                  {name}
                </div>
            </button>
              
    )
}

export default ChannelMenuItem