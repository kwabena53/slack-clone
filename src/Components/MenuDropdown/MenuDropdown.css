

.close{
    display: none;
}

.dropdownMenuIcon:hover {
    background: rgb(255, 255, 255, 0.2);
    border-radius: 3px;
}

.dropdownMenuIcon:focus {
    transform: rotate(90deg);
}