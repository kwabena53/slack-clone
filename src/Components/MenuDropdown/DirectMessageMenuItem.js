
import '../../App.css'
import './MenuDropdown.css'


const DirectMessageMenuItem = ({user, ...rest})=>{
  const onlineStatus = user?.active? "active": "inactive"
    return(
            <button {...rest} className="menuItem">
                <div className="subMenu">
                  <div className="menuIcon">
                      {/* <div className="imgIcon"> */}
                      <div className="dmIcon">
                      <img className="imgIcon" src={user?.url}/>
                        <div className={`online-icon ${onlineStatus}`}></div>
                      </div>
                      {/* </div> */}
                    </div>
                  {user?.name}
                </div>
            </button>
              
    )
}

export default DirectMessageMenuItem