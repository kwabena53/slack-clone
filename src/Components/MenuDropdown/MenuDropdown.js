
import '../../App.css'
import './MenuDropdown.css'
import React, {useState} from 'react';
import {ArrowRightBold} from '../../Components/Icons/SlackCloneIcons';
const MenuDropdown = ({title})=>{

    const [show, setShow] = useState(true)

   const handleMenuClick = ()=>{
        setShow(!show)
    }
    
    let active = show ? "" : "close"
    
    return(
        <>
        <div className="menuParent">
              <div className="navItem">
                <div> <button onClick={handleMenuClick}  className="dropdownMenuIcon dpIcon"><ArrowRightBold className="navIcon" height="1em" /> </button> </div>
                <div  >{title}</div>
              </div>
        </div>
        <div className={`subContainer ${active}`} >
        <button className="menuItem">
                <div className="subMenu">
                  <span className="menuIcon"> #</span>
                  Announcement
                </div>
              </button>
              <button className="menuItem">
                <div className="subMenu">
                  <span className="menuIcon"> #</span>
                  community
                </div>
              </button>
              <button className="menuItem">
                <div aria-selected="true" aria-posinset="2" role="treeitem" aria-level="2" className="subMenu">
                  <span className="menuIcon"> #</span>
                  ms-information-technology
                </div>
            </button>
        </div>
                
        </>
    )
}

export default MenuDropdown