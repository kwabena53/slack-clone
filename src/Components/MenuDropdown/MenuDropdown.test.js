

import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import ChannelMenuItem from "./ChannelMenuItem";
import MenuDropdown from "./MenuDropdown";



test("renders right MenuDropdown", () => {
  render(<MenuDropdown  children={<ChannelMenuItem name="new hires"/>} title="Channel" />);
  const textElement = screen.getByText(/Channel/i);
  expect(textElement).toBeInTheDocument();
});

test("matches MenuDropdown snapshot", () => {
  const component = renderer.create(<MenuDropdown  children={<ChannelMenuItem name="new hires"/>} title="Channel" />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

