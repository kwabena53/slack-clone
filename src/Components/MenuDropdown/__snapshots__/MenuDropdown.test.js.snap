// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches MenuDropdown snapshot 1`] = `
Array [
  <div
    className="menuParent"
  >
    <div
      className="navItem"
    >
      <div
        className="navHeader"
      >
         
        <button
          className="dropdownMenuIcon dpIcon "
          onClick={[Function]}
        >
          <svg
            className="navIcon"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.9188 8.17969H11.6888H6.07877C5.11877 8.17969 4.63877 9.33969 5.31877 10.0197L10.4988 15.1997C11.3288 16.0297 12.6788 16.0297 13.5088 15.1997L15.4788 13.2297L18.6888 10.0197C19.3588 9.33969 18.8788 8.17969 17.9188 8.17969Z"
            />
          </svg>
           
        </button>
         
      </div>
      <div
        className="menuTitle"
      >
        Channel
      </div>
      <div
        className="leftIcons"
      >
        <div
          className="leftIcon dropdownMenuIcon dpIcon"
        >
           
          <svg
            className="navIcon"
            enableBackground="new 0 0 24 24;"
            fill="white"
            height="1em"
            id="Layer_1"
            stroke="white"
            version="1.1"
            viewBox="0 0 24 24"
            width="1em"
            x="0px"
            xmlSpace="preserve"
            xmlns="http://www.w3.org/2000/svg"
            xmlnsXlink="http://www.w3.org/1999/xlink"
            y="0px"
          >
            <circle
              cx="12"
              cy="3"
              r="1.1"
            />
            <circle
              cx="12"
              cy="12"
              r="1.1"
            />
            <circle
              cx="12"
              cy="21"
              r="1.1"
            />
          </svg>
           
        </div>
        <div
          className="leftIcon dropdownMenuIcon dpIcon"
        >
          <svg
            className="navIcon"
            fill="none"
            height="1.2em"
            viewBox="0 0 24 24"
            width="1.2em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 12H18"
              stroke="#ffffff"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="1.5"
            />
            <path
              d="M12 18V6"
              stroke="#ffffff"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="1.5"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="subContainer "
  >
    <button
      className="menuItem"
    >
      <div
        className="subMenu"
      >
        <span
          className="menuIcon"
        >
           #
        </span>
        new hires
      </div>
    </button>
  </div>,
]
`;
